packages/salesforcedx-vscode-apex/src/languageServer.ts

```diff
@@ -148,20 +168,16 @@ export async function createLanguageServer(
     uriConverters: {
       code2Protocol: code2ProtocolConverter,
       protocol2Code: protocol2CodeConverter
-    }
+    },
+    initializationOptions: {
+      detectEmbeddedSoqlCompletion: soqlExtensionInstalled
+    },
+    ...(soqlExtensionInstalled ? { middleware: soqlMiddleware } : {})
   };
+}
 
-  const server = await createServer(context);
-  const client = new LanguageClient(
-    'apex',
-    nls.localize('client_name'),
-    server,
-    clientOptions
-  );
-
-  client.onTelemetry(data =>
-    telemetryService.sendEventData('apexLSPLog', data.properties, data.measures)
-  );
-
-  return client;
+function isSOQLExtensionInstalled() {
+  const soqlExtensionName = 'salesforce.salesforcedx-vscode-soql';
+  const soqlExtension = vscode.extensions.getExtension(soqlExtensionName);
+  return soqlExtension !== undefined;
```

Comment:
Let's also check if the extension is enabled, since you can have it installed but disabled. I believe it's just `soqlExtension.isEnabled`.

Reply:
When the extension is disabled, `getExtension` returns `undefined`. There's no `isEnabled`, although there might be in the future: https://github.com/microsoft/vscode/issues/84997

I did try with `.isActive` like you suggested on the quip doc: unfortunately, the SOQL extension is not yet activated when this check runs.

Reply:
ah i see, thanks for clarifying.